### DISCUSSION TITLE
How to control buttons' state and content?

### DISCUSSION THREAD

**Discussion by marospekarik** [pmndrs/leva#443] [Q&A]
Hi,

I am using a button helper and struggling to understand a few things. How do I change the label and set the button disabled programmatically? 

`  const leftPanel = useControls(
      {
          button: button(() => {{sendMessage('hello')}}, { disabled: false}),
      },
      { store: myStore }
  )`

I see that documentation or examples for this are missing. I've tried to use something like `myStore.set(), or destruct the object, however, it doesn't work for the button helper.

**Comment by dbismut** [comment]
@marospekarik, you can use a deps array as the last argument for useControls, the schema recompute whenever the deps change.

**Reply by marospekarik** [reply]
But how would I change the button's label for example? I would need to replace the whole entry, right?

**Reply by dbismut** [reply]
Well the label value would depend on the deps.

**Reply by marospekarik** [reply]
I am still struggling to wrap my head around this. How would you disable the button after the initial state? I am trying:

`storeLeft.set({ ["init"]: { disabled: true } }, false);`

replacing it completely:
`storeLeft.set({ ["init"]: button(() => {sendMessage()}, { disabled: true })}, false);`

But nothing really works
`

**Reply by dbismut** [reply]
Have a look at this https://codesandbox.io/s/suspicious-bessie-lo6dx3?file=/src/App.js

**Comment by marospekarik** [comment]
Thanks @dbismut , that's what I was looking for!

**Comment by marospekarik** [comment]
Unfortunately, there is still something wrong with all of this, when the button turns to disabled mode: 

`
  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
  const [controls, setControls, get]  = useControls( () => ({
          [isButtonDisabled ? "Loading...": "Initialize"]: button(() => 
            sendMessage(), 
            {disabled: isButtonDisabled}
          ),
      }
  ), { store: storeLeft }, [isButtonDisabled]);
`

![image](https://user-images.githubusercontent.com/14806065/222149117-bf887495-f589-4fc1-9223-e1aea726da60.png)

**Reply by dbismut** [reply]
What am I looking at?

**Reply by marospekarik** [reply]
Somehow after when the state change to true, there is some strange hierarchy. See these two images:
1. isButtonDisabled  = false
![image](https://user-images.githubusercontent.com/14806065/222153189-c7d24ff9-b12a-4f10-a026-7634c2317837.png)
2. isButtonDisabled  = true
![image](https://user-images.githubusercontent.com/14806065/222153442-79a23d90-766b-470e-b74f-8e641ceaab19.png)

**Comment by marospekarik** [comment]
See here @dbismut : https://codesandbox.io/s/elated-kare-pwmzit?file=/src/App.js

**Reply by dbismut** [reply]
I don't get it

**Reply by marospekarik** [reply]
Lol... it's because of the three dots in the name of the button (...)

**Reply by dbismut** [reply]
Ooooh right yes Leva uses dot notation to manage folders nesting for keys. You can try using special character …, or otherwise use the option "label".

**Reply by marospekarik** [reply]
> Ooooh right yes Leva uses dot notation to manage folders nesting for keys. You can try using special character …, or otherwise use the option "label".

Thanks for the clarification, how to use the option "label"? This for example doesn't work for me:

`[isButtonDisabled ? "Loading…": "Initialize"]: button(() => 
    sendMessage(), 
    {disabled: isButtonDisabled, label: "myLabel"}
  ),
`

**Reply by dbismut** [reply]
If it doesn't work then it hasn't been properly implemented for the button unfortunately, it's a bug.

**Comment by geyang** [comment]
The order of the button also gets messed up.

It moved to the end.
